packages/simulation/tests/convertNFAtoDFA.test.ts

```diff
@@ -156,4 +157,51 @@ describe('Check to ensure DFA graph is displayed as expected', () => {
     expect(graph.transitions.filter((transition) => transition.from === 5)[1].to).toBe(5)
     expect(graph.transitions.filter((transition) => transition.from === 5)[1].read).toBeOneOf(['A', 'B'])
   })
+  test('Graph should use a single trap state instead of multiple when converted to a DFA', () => {
+    const graph = reorderStates(convertNFAtoDFA(reorderStates(convertSingleTrapState as any) as any) as any)
+    // Initial state should be q0
+    expect(graph.initialState).toBe(0)
+    // They would be 6 if they got returned as a DFA rather than 4 as an NFA
+    expect(graph.states.length).toBe(4)
+    // Should be two final states, q2 and q3
+    expect((graph.states.find((state) => state.id === 0)).isFinal).toBe(false)
+    expect((graph.states.find((state) => state.id === 1)).isFinal).toBe(false)
+    expect((graph.states.find((state) => state.id === 2)).isFinal).toBe(false)
+    expect((graph.states.find((state) => state.id === 3)).isFinal).toBe(true)
+    // No lambda transition should be found, and should be 28 transitions total. All states should have a single transition 'A' to 'G' and be connected in a certain way
+    expect(graph.transitions.length).toBe(28)
+    expect(graph.transitions.some((transition) => transition.read === undefined)).toBe(false)
+    expect(graph.transitions.every((transition) => transition.read === 'A' || transition.read === 'B' || transition.read === 'C' || transition.read === 'D' || transition.read === 'E' || transition.read === 'F' || transition.read === 'G')).toBe(true)
```

Comment:
```suggestion
    const alphabet = 'ABCDEFG'
    expect(graph.transitions.every((transition) => alphabet.includes(transition.read))).toBe(true)
```
Bit redundant, you can just test both at once